src/schematools/contrib/django/factories.py

```diff
@@ -487,5 +519,125 @@ def model_factory(
+            try:
+                # Also get rid of the `shortname`
+                del field_definition["shortname"]
+            except KeyError:
+                pass
+            # convert object-type fields to strings
+            if field.type == "object":
+                field_definition["type"] = "string"
+                del field_definition["properties"]
+
+            # Add autoincrementing behaviour to the identifier fields
+            # as a hint during selection of a proper faker factory
+            field_definition["type"] = field_definition["type"] + "/autoincrement"
+
+        new_table_data["schema"]["properties"][f"{field_name}{id_post_fix}"] = field_definition
+    return DatasetTableSchema(new_table_data, parent_schema=table_schema._parent_schema)
+
+
+def model_mocker_factory(
+    dataset: Dataset, table_schema: DatasetTableSchema, base_app_name: Optional[str] = None
+) -> Type[DynamicModelMocker]:
+    """Generate a Django model mocker class from a JSON Schema definition."""
+    dataset_schema = dataset.schema
+    app_label = f"{dataset_schema.id}"
+    base_app_name = base_app_name or "dso_api.dynamic_api"
+
+    # Bootstrap for the model_factory is implemented in the DynamicRouter, so on startup
+    # of the DSO-API. We have no desire to bootstrap the DynamicModelMocker for DSO runtime,
+    # just in the test suite and in the relevant management commands. As the DynamicModelMocker
+    # wraps around the DynamicModel, we must initiate it here to feed to the DjangoModelFactory
+    # Meta class:
+    # https://factoryboy.readthedocs.io/en/stable/orms.html#the-djangomodelfactory-subclass.
+    # This table_schema is stripped from relation info, we want to use simple
+    # scalars for relations.
+    stripped_table_schema = _simplify_table_schema_relations(table_schema)
+    model_factory(dataset, stripped_table_schema, base_app_name=base_app_name)
+
+    # Generate fields
+    fields = {}
+    for field in stripped_table_schema.fields:
+        type_ = field.type
+        # skip schema field for now
+        if type_.endswith("definitions/schema"):
+            continue
+        # skip nested tables and fields that are only added for temporality
+        if field.is_nested_table or field.is_temporal:
+            continue
+        # reduce amsterdam schema refs to their fragment
+        if type_.startswith(settings.SCHEMA_DEFS_URL):
+            type_ = urlparse(type_).fragment
+
+        # Generate name, fix if needed.
+        field_name = to_snake_case(field.name)
+
+        # If, in addition to a type, a format has been defined
+        # for a field, the format will be used to look up the
+        # appropriate provider.
+        if (format_ := field.format) is not None:
+            type_ = format_
+
+        # If a faker has been defined for the field, this
+        # faker is used for the provider lookup.
+        if (faker := field.faker) is not None:
+            type_ = faker
```

Comment:
Dit kan vast wat netter :-)

Reply:
Deze stond idd op mijn lijstje om te refactoren in een aparte method. De mocker is nog een beetje een eerste versie op die moment die nog op allerlei punten moet worden verfijnd.

Reply:
Ik ga dit refactoren, de logica verplaatsen naar de faker implementatie. Voor nu laat ik de code even zoals deze is.